src/main/java/techreborn/init/ModPlates.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * This file is part of TechReborn, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) 2018 TechReborn
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package techreborn.init;
+
+import net.minecraft.item.Item;
+import net.minecraft.util.IStringSerializable;
+import techreborn.lib.ModInfo;
+import techreborn.items.ItemPlates;
+
+
+/**
+ * @author drcrazy
+ *
+ */
+public enum ModPlates implements IStringSerializable {

```

Comment:
Looks like a good start.

Would it be possible to set the camel case name instead of having to duplicate it twice?

`ADVANCED_ALLOY_PLATE("plateAdvancedAlloy"),`

would then just be

`ADVANCED_ALLOY_PLATE,`

Reply:
Options I've considered:
1) "PLATE_ADVANCED_ALLOY," this could be easily case converted to oreDict,
2) "ADVANCED_ALLOY,", this could be prefixed and case converted to oreDict 

Moreover, since not a lot left in ItemPlates class maybe we will end up with ItemMaterial, which will accept allowed prefixes either in constructor or via hasPlate("true") method.

Reply:
I think just `ADVANCED_ALLOY` would be good, as when referencing it it would be `ModPlates.ADVANCED_ALLOY` no need to duplicate the plate bit. 

If you did extra data for them you could have a plates.json file somewhere that has stuff like the oredict name and what not. (Thinking about it is not a great idea)

But it would be really nice to have everything in oneplace.